samples/v1/types_config.go

```diff
@@ -0,0 +1,212 @@
+package v1
+
+import (
+	operatorv1 "github.com/openshift/api/operator/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// Config contains the configuration and detailed condition status for the Samples Operator.
+type Config struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata" protobuf:"bytes,1,opt,name=metadata"`
+
+	// +kubebuilder:validation:Required
+	// +required
+	Spec ConfigSpec `json:"spec" protobuf:"bytes,2,opt,name=spec"`
+	// +optional
+	Status ConfigStatus `json:"status,omitempty" protobuf:"bytes,3,opt,name=status"`
```

Comment:
needs godoc: ex: `// status contains the observed configuration and other details describing the state of the Samples Operator.`